DESIGN-297: define browser support

klepas opened the pull request:
## Description

Adds a `BROWSER_SUPPORT.md` document in the project root to provide a 1-pager on the UI-Kit’s browser support strategy.
## Additional information

I’m confident that we have a general seal of approval by:
- Andrew Arch
- Dan Draper
- Dan Whitmarsh/D7
## Definition of Done
- [x] Content/documentation reviewed by Julian or someone in the Content Team
- [x] Proposal reviewed by Gary or someone the Design team
- [x] Proposal reviewed by all of the core developers
- [x] Stakeholder/PO review
- [x] CHANGELOG updated

Do we have any DoD communication tasks related to this PR, viz., communicating our stance upwards or laterally?

TrebBrennan committed:
File setup and draft

Commit:
Remove Tests section from Browser Support

Commit:
Update full/basic support level of IE8 & IE9

Commit:
Improvements to doco

Can take another look when the list is completed @petronbot

Commit:
Some further edits and ideas on the browser support draft doc.

Commit:
Clarify meaning of support.

Commit:
Fleshes out browser support doc post chat with D7 and Andrew Arch.

elisechant commented:
![screen shot 2016-08-19 at 11 49 10 am](https://cloud.githubusercontent.com/assets/2195815/17796509/490f1e32-6604-11e6-9d73-925891fd8195.png)

Generally I agree with the drafted browser-support guideline you’ve put together other than the definition of “Support: Full” - this is not really Progressive Enhancement. And I also think there should be at least some mention about Mobile or Touch device support.

klepas commented:
> […] other than the definition of “Support: Full” - this is not really Progressive Enhancement. And I also think there should be at least some mention about Mobile or Touch device support.

Agreed. I didn’t like the terminology of ‘basic’ support (hence ‘functional’); and conversely, ‘full’ support has a similar issue. I agree that this isn’t PE either.

I’m also in agreement that we must include some information on mobile device support, considering we should be mobile-first.

ashnewport commented:
Just a thought: is there always the intention to test and support from the noted browser versions and up? Would that mean modifying the support list would then require a major(?) version release?

I personally like how [jQuery](https://jquery.com/browser-support/) and [Bootstrap](http://getbootstrap.com/getting-started/#support) approach their browser support.

Seems like the above approaches could help to mean less frequent updating of the supported versions list.

lhowdin commented:
I think we also need to mention that not only do we want to do this, but us and the people that will use the kit are actually obligated 'to support all users according to their needs, regardless of user agent or browser technologies.'  Andrew Arch will have the details of the law/mandates that apply.

lhowdin commented:
The Disability Discrimination Act says equal access to information to information on laws and government programs.

D.D.A. guide: A responsible government

The Federal Disability Discrimination Act (D.D.A.) makes disability discrimination against the law in many areas of life.

This includes provision of services of a kind provided by government. All governments in Australia - Commonwealth, State, Territory and local government - have responsibilities under the D.D.A.

This includes:

Access to places where government programs are being run.
Equal access to information on laws and government programs.
Access to voting places, facilities and information
Access to court buildings

Commit:
Content shaping

I’ve done a fairly sizeable edit to pair back to basics. Happy to chat
with you @klepas about what needs to go back in.

joolswood commented:
I've just done an edit. Hopefully I'm progressing on at least some of the points here.

A couple of comments that aren't yet reflected in my commit(s):
- We're using similar principles on the home page and elsewhere in the doco. A specific principles section is one thing Gary and I talked about a while ago. It's probably out of scope for this sprint, but I think we should be agreeing on this as a Guide + KIT wide thing and then making its over-arching role better reflected in the structuring of the doco (something to discuss in our vision workshop @hannah-ustwo).
- I was tending to think of the 'fully supported' level as defined by 'advanced' functionality. I'm not sure if 'advanced' is the right wording/concept, but this might be a step more in the direction of what you're talking about @elisechant ?

AndrewArch commented:
Yes, we do have an obligation to support all users, but not all them will be on line, and some them maybe not be appropriately online. I suspect that providing support for Mosaic for example would conflict with the Procurement Rules of [value-for-money](https://www.finance.gov.au/procurement/procurement-policy-and-guidance/commonwealth-procurement-rules/cprs-value-for-money.html) - and where does something like Lynx fit?. We do need to settle on a reasonable compromise, and be certain that the offline channels do not compromise the service for those that can't do digital (or need to step out of the digital process for whatever reason).

Commit:
Minor edits following on from Jools.

Commit:
Links to DDA 1992 + table formatting improvs.

Commit:
Links to latest DDA.

klepas commented:
I’ve edited this a bit further to reflect some of the legal obligations.

@joolswood, @petronbot, @hannah-ustwo, @danwhitmarsh, @AndrewArch,  could I ask for a review? (:

Commit:
Edits

Some inline comments to come.

joolswood commented on BROWSER-SUPPORT.md:
```diff
@@ -1,7 +1,5 @@
 # Browser support
 
-## Introduction
```
I removed the extra heading as by definition the first content should introduce the thing.

joolswood commented:
I'm uncertain about line 42 @klepas - `JavaScript and CSS not necessarily required.` Doesn't our principle at line 55 commit us to not requiring it? I think I get your meaning but it may over complicate doing it this way.

Also, WRT to line 50; do we need to be more specific about `Advanced design` - following on from above point, should we talk about CSS here instead.

joolswood commented on BROWSER-SUPPORT.md:
```diff
@@ -1,7 +1,5 @@
-
 The UI-Kit team needs to support all of our users, regardless of their device, web browser or other user agent.
 
 Alongside accessibility, extensive browser and device support is a legal requirement under the <a href="https://www.legislation.gov.au/Latest/C2016C00763" rel="external">Disability Discrimination Act (1992)</a>, requiring equal access to information on laws and government programs.
```
I'm uncertain about the value of this para @klepas I'm all for promoting our obligation to accessibility (I made sure we start off by saying this, albeit in less direct terms, at line 5).

If we want to add value (using a reference to our legal obligation) it should be provided alongside direction for kit users to implement the kit in an accessible fashion, which is probably best situated elsewhere in the guide.

Also I reckon there's value in being more direct about this in our overall principles, but they should also live elsewhere.

klepas replied:
I added this since we’ve referenced our _accessibility obligations_ previously, whereas the browser support philosophy/policy we’re doing here — I feel —overlaps, but also extends on our accessibility obligations.

I guess I see this as a guiding standard/policy document, and ref’ing the Act that informs our requirements seemed natural to me?

Worth noting that this document should be a page/section/… somewhere within the Design Guide when we move away from KSS.

joolswood replied:
What about moving the reference to the Act to the README.md in the meantime? I can rework the text here to link back to it. How about have a chat tomorrow? @klepas

klepas replied:
*Nod\* Works with me. (:

joolswood commented:
Please see my inline comment about the legislation ref.

klepas commented:
> Please see my inline comment about the legislation ref.

Added an inline comment on this. (:

> I'm uncertain about line 42 @klepas - JavaScript and CSS not necessarily required. Doesn't our principle at line 55 commit us to not requiring it? I think I get your meaning but it may over complicate doing it this way.

Ah, good point. I think we might want to amend that earlier line then, since some of our legacy browser support will require shims/polyfills/other JS and/or CSS hacks to get older browsers working… of course not always the case.

> Also, WRT to line 50; do we need to be more specific about Advanced design - following on from above point, should we talk about CSS here instead.

Yea, good idea.

Commit:
Quick fixes

@petronbot can you please check these changes - they are in response to
the comments between @klepas and I at bottom of
https://github.com/AusDTO/gov-au-ui-kit/pull/288

Commit:
Small edit to @klepas text

Commit:
Split up Browser support tables into 3 sections

Commit:
Specify version 'IE10+' in Browser support

Commit:
Add 'Tested' status column to IE versions in Browser Support

Commit:
Combine Desktop and IE browser support tables

Commit:
A few tweaks for clarity

Thanks @petronbot
